src/libraries/Common/src/System/Text/Json/PooledByteBufferWriter.cs

```diff
@@ -22,6 +22,12 @@ internal sealed class PooledByteBufferWriter : IBufferWriter<byte>, IDisposable
 
         private const int MinimumBufferSize = 256;
 
+#if NETCOREAPP
+        private static readonly int MaximumBufferSize = Array.MaxLength;
+#else
+        private const int MaximumBufferSize = int.MaxValue;
```

Comment:
Should this simply hardcode the array max value instead of using the largest integer?

Reply:
Done, I've copied the value from Array.cs.